x-pack/plugins/observability/public/context/co_pilot_context/create_co_pilot_service.ts

```diff
@@ -18,94 +22,136 @@ function getMessageFromChunks(chunks: CreateChatCompletionResponseChunk[]) {
   return message;
 }
 
-export function createCoPilotService({ enabled, http }: { enabled: boolean; http: HttpSetup }) {
+export function createCoPilotService({
+  enabled,
+  trackingEnabled,
+  http,
+}: {
+  enabled: boolean;
+  trackingEnabled: boolean;
+  http: HttpSetup;
+}) {
   const service: CoPilotService = {
     isEnabled: () => enabled,
+    isTrackingEnabled: () => trackingEnabled,
```

Comment:
why does this need to be a function? Can't you just set the value of `trackingEnabled` to isTrackingEnabled?

* same for `isEnabled`

Reply:
just future-proofing, that's it. the alternative is a readonly property or a `get isTrackingEnabled` which are not super appealing either (for a variety of reasons)